core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -411,13 +419,26 @@ public class ReaderBuilder extends FileHelper implements ReaderTableConfiguratio
     private CacheProvider cacheProvider;
     private Cache<String,Long> fileLenCache;
     private boolean seekToBeginning = false;
+    private Range fence;
 
     public ReaderTableConfiguration forFile(String filename, FileSystem fs, Configuration fsConf,
         CryptoService cs) {
       filename(filename).fs(fs).fsConf(fsConf).cryptoService(cs);
       return this;
     }
 
+    public ReaderTableConfiguration forFile(TabletFile file, FileSystem fs, Configuration fsConf,
+        CryptoService cs) {
+      return forFile(file.getPathStr(), file.getFence(), fs, fsConf, cs);
+    }
+
+    public ReaderTableConfiguration forFile(String filename, Range fence, FileSystem fs,
```

Comment:
Would creating a `withRange` method reduce the number of overall changes? Calling code could be:
```
forFile(file, fs, conf, cs).withRange(range)
```

Reply:
Yeah I can rework it a bit and add that, it would probably make things a little better with sticking to the builder pattern.